jar-infer/scripts/android-jar.py

```diff
@@ -20,7 +20,9 @@
 
 ### Finding android libraries ###
 print "> Finding android libraries..."
-cmd = "find " + ajars_dir + " -name \"*.jar\""
+# Since SDK 31, the directory structure mixes classes.jar and classes-header.jar files, we use
+# only the former, as classes-header.jar files are missing methods' bytecode.
+cmd = "find " + ajars_dir + " -name \"*.jar\" | grep -v classes-header.jar"
```

Comment:
Not for this PR necessarily, but probably this directory-searching logic could just be implemented in Python rather than using a subprocess and the code might be clearer.  Maybe we can add a TODO comment

Reply:
This script already calls a lot of commands through `Popen(cmd, stdout=subprocess.PIPE, shell=True)`. See the various calls to `jar tvf ... | grep ... | awk ...` below this line. It is also part of our release process only (i.e. not run or shipped to NullAway/JarInfer users). 

I am not opposed to switching this logic to a directory traversal in python, but I am also not sure it will make any difference for the overall quality of the script. This is basically a shell script with a dictionary 😄

Reply:
Yes, you are right.  Fair enough 🙂